File: 0118-pascals-triangle/0118-pascals-triangle.js
/**
 * @param {number} numRows
 * @return {number[][]}
 */
var generate = function(numRows) {
  const answer = [[1]]
  
  for(let i = 1; i < numRows; i++){
    const row = []
    for(let j = 0; j <= i; j++){
      if(j===0 || j===i){
        row.push(1)
      } else {
        row.push(answer[i-1][j-1]+answer[i-1][j])
      }
    }
    answer.push(row)
  }
  return answer
};

File: 0118-pascals-triangle/0118-pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> dp = new ArrayList<>();
    
        for(int i = 0; i < numRows; i++){
            List<Integer> row = new ArrayList<>(Collections.nCopies(i+1, 1));
            for(int j =1; j < i; j++){
                row.set(j, dp.get(i-1).get(j-1) + dp.get(i-1).get(j));
            }
           dp.add(row);
        }
        
        return dp;
    }
}

